### ISSUE TITLE
False positive: Unexpected keyword argument 'allow_pickle' for numpy.load

### ISSUE THREAD

**Issue by thawn** [PyCQA/pylint#3567]
### Steps to reproduce
1. code example:
```python
import numpy
data = numpy.load('test.npz', allow_pickle=False)
```

2. run pylint 

### Current behavior
pylint complains: `Unexpected keyword argument 'allow_pickle' in function call pylint(unexpected-keyword-arg)`

### Expected behavior
pylint should not complain about the `allow_pickle` keyword argument for `numpy.load`.
 
However, it could be good to issue a verbose security warning (but not an error) if `allow_pickle` is set to `True` because this poses a security risk [as explained in the numpy documentation](https://numpy.org/doc/stable/reference/generated/numpy.load.html?highlight=load#numpy.load).

### pylint --version output
pylint 2.5.0
astroid 2.4.0
Python 3.7.5 (default, Oct 25 2019, 15:51:11) 
[GCC 7.3.0]
numpy 1.17.3

**Comment by hippo91** [comment]
@thawn thanks for the report.
In ``numpy`` version prior to 1.18.0 there are two functions `load`. One in the `numpy.lib.npyio` module, which is the one you use, and on in the `numpy.core.numeric` module which accepts only one argument. 
`astroid`, is the inference engine used by `pylint` and doesn't do flow control for now. Thus it is impossible for it to determine which of those functions you use, that's why you have this false positive message.

To my mind, you can work around this by :
- moving to `numpy` 1.18 or above;
- import more specifically the module you use:
```
import numpy.lib.npyio as npyio
data = npyio.load('test.npz', allow_pickle=False)
```
